app/Http/Controllers/Admin/CategoriesController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Models\Meeting;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use Illuminate\Support\Facades\Storage;
+
+class CategoriesController extends Controller
+{
+    const ICON_FOLDER = 'image/category/';
+
+    public function index()
+    {
+        $all_categories = Category::withTrashed()->paginate(10);
+        return view('admin.chatrooms.categories.index')
+            ->with('all_categories', $all_categories);
+    }
+    public function show($id, Meeting $meeting)
+    {
+        // update Meetings Status
+        $meeting->updateStatus();
+        
+        $category = Category::withTrashed()->findOrFail($id);
+        $meetings = $category->meetings()->withTrashed()->latest()->paginate(10);
+        return view('admin.chatrooms.categories.show')
+            ->with('category', $category)
+            ->with('meetings', $meetings)
+            ->with('statusColor', $meeting->statusColor());
+    }
+    public function add()
+    {
+        return view('admin.chatrooms.categories.add');
+    }
+    public function edit($id)
+    {
+        $category = Category::withTrashed()->findOrFail($id);
+        return view('admin.chatrooms.categories.edit')
+        ->with('category', $category);
+    }
+
+    public function store(Request $request, Category $category)
+    {
+        $request->validate([
+            'name'          => 'required|max:20|unique:categories,name',
+            'color'         => 'required|regex:/^#[0-9a-fA-F]{3}([0-9a-fA-F]{3})?$/', //Hex Color
+            'icon'          => 'required|mimes:jpg,jpeg,png,gif',
+            'description'   => 'required|min:5|max:100'
+        ]);
+
+        $category->name = str_replace(" ", "_", ucwords($request->name));
```

Comment:
sine you remove the category from the parameters, here, you can just save category like ...
```
Category::create([
    'color' => $request->color,
    'descrition' => $request->description,
    
    // add all other field here....
]);
```

Reply:
I changed it